File: solutions/elixir/bird-count/1/lib/bird_count.ex
defmodule BirdCount do
  def today([]) do
    nil
  end

  def today([today_count | _]) do
    today_count
  end

  def increment_day_count([]) do
    [1]
  end

  def increment_day_count([today_count | rest_count_lst]) do
    [today_count + 1 | rest_count_lst]
  end

  def has_day_without_birds?(list) do
    0 in list
  end

  def total(list) do
    Enum.sum(list)
  end

  def busy_days(list) do
    list
    |> Enum.filter(fn day_count -> day_count >= 5 end)
    |> Enum.count()
  end
end

File: solutions/ruby/bird-count/1/bird_count.rb
class BirdCount

  def self.last_week
    [0, 2, 5, 3, 7, 8, 4]
  end

  def initialize(birds_per_day)
    @birds = birds_per_day
  end

  def yesterday
    @birds[-2]
  end

  def total
    @birds.sum
  end

  def busy_days
    @birds.filter {|e| e >= 5}.count
  end

  def day_without_birds?
    @birds.filter {|e| e == 0}.any?
  end
end
